src/Microsoft.PowerShell.Commands.Management/commands/management/NewPropertyCommand.cs

```diff
@@ -175,4 +182,118 @@ protected override void ProcessRecord()
+    {
+        private static readonly string[] s_RegistryPropertyTypes = new string[]
+        {
+            "String",
+            "ExpandString",
+            "Binary",
+            "DWord",
+            "MultiString",
+            "QWord",
+            "Unknown"
+        };
+
+        private static string GetRegistryPropertyTypeToolTip(string propertyTypeName) => propertyTypeName switch
+        {
+            "String" => TabCompletionStrings.RegistryStringToolTip,
+            "ExpandString" => TabCompletionStrings.RegistryExpandStringToolTip,
+            "Binary" => TabCompletionStrings.RegistryBinaryToolTip,
+            "DWord" => TabCompletionStrings.RegistryDWordToolTip,
+            "MultiString" => TabCompletionStrings.RegistryMultiStringToolTip,
+            "QWord" => TabCompletionStrings.RegistryQWordToolTip,
+            _ => TabCompletionStrings.RegistryUnknownToolTip
+        };
+
+        private static readonly char[] QuoteChars = new char[]
+        {
+            '\'',
+            '"',
+            '\u2018', // left single quotation mark
+            '\u2019', // right single quotation mark
+            '\u201a', // single low-9 quotation mark
+            '\u201b', // single high-reversed-9 quotation mark
+            '\u201c', // left double quotation mark
+            '\u201d', // right double quotation mark
+            '\u201E'  // low double left quote used in german
+        };
+
+        /// <summary>
+        /// Returns completion results for PropertyType parameter.
+        /// </summary>
+        /// <param name="commandName">The command name.</param>
+        /// <param name="parameterName">The parameter name.</param>
+        /// <param name="wordToComplete">The word to complete.</param>
+        /// <param name="commandAst">The command AST.</param>
+        /// <param name="fakeBoundParameters">The fake bound parameters.</param>
+        /// <returns>List of Completion Results.</returns>
+        public IEnumerable<CompletionResult> CompleteArgument(
+            string commandName,
+            string parameterName,
+            string wordToComplete,
+            CommandAst commandAst,
+            IDictionary fakeBoundParameters)
+        {
+            // -PropertyType parameter is only supported on Windows
+            if (!Platform.IsWindows)
+            {
+                yield break;
+            }
+
+            Collection<PathInfo> paths;
+
+            if (fakeBoundParameters.Contains("Path"))
+            {
+                paths = ResolvePath(fakeBoundParameters["Path"], isLiteralPath: false);
+            }
+            else if (fakeBoundParameters.Contains("LiteralPath"))
+            {
+                paths = ResolvePath(fakeBoundParameters["LiteralPath"], isLiteralPath: true);
+            }
+            else
+            {
+                paths = ResolvePath(@".\", isLiteralPath: false);
+            }
+
+            if (paths[0].Provider.NameEquals("Registry"))
```

Comment:
Why don't we check paths.Count > 0?

Reply:
Ah I must have forgot to put that back when I was refactoring.